simulator/src/main/java/com/hazelcast/simulator/test/TestContainer.java

```diff
@@ -15,409 +15,281 @@
-                    int intervalMillis = getMetronomeIntervalMillis(field, runWithWorkerMetronomeInterval);
-                    MetronomeType type = getMetronomeType(field, runWithWorkerMetronomeType);
-                    injectMap.put(field, withFixedIntervalMs(intervalMillis, type));
-                }
-            }
-            classType = classType.getSuperclass();
-        } while (classType != null);
-        return injectMap;
-    }
-
-    private Map<Enum, Probe> createOperationProbeMap(Class<? extends IWorker> workerClass, IWorker worker) {
-        if (!IMultipleProbesWorker.class.isAssignableFrom(workerClass)) {
-            return null;
+        if (runWithWorker != null) {
+            runStrategies.add(new RunWithWorkersRunStrategy(this, runWithWorker));
         }
 
-        // remove the default worker probe
-        probeMap.remove(DEFAULT_WORKER_PROBE_NAME);
+        List<Method> timeStepMethods = new AnnotatedMethodRetriever(testClass, TimeStep.class)
+                .findAll();
 
-        Map<Enum, Probe> operationProbes = new HashMap<Enum, Probe>();
-        for (Enum operation : ((IMultipleProbesWorker) worker).getOperations()) {
-            String probeName = capitalizeFully(operation.name(), '_').replace("_", "") + "Probe";
-            operationProbes.put(operation, getOrCreateProbe(probeName, true));
+        if (!timeStepMethods.isEmpty()) {
+            runStrategies.add(new TimeStepRunStrategy(this));
         }
-        return operationProbes;
-    }
 
-    private Probe getOrCreateProbe(String probeName, boolean partOfTotalThroughput) {
-        Probe probe = probeMap.get(probeName);
-        if (probe == null) {
-            probe = (runWithWorkerIsLightweightProbe
-                    ? new ThroughputProbe(partOfTotalThroughput)
-                    : new HdrProbe(partOfTotalThroughput));
-            probeMap.put(probeName, probe);
+        if (runStrategies.isEmpty()) {
+            throw new IllegalTestException(format("Test must contain either %s or %s or %s method",
+                    Run.class, RunWithWorker.class, TimeStep.class));
         }
-        return probe;
-    }
 
-    private IWorker runWorkers(int threadCount, Method runMethod, Map<Field, Object> injectMap,
-                               Map<Enum, Probe> operationProbes) throws Exception {
-        IWorker firstWorker = null;
-        ThreadSpawner spawner = new ThreadSpawner(testContext.getTestId());
-        for (int i = 0; i < threadCount; i++) {
-            final IWorker worker = invokeMethod(testClassInstance, runMethod);
-            if (firstWorker == null) {
-                firstWorker = worker;
-            }
-
-            injectObjects(injectMap, worker);
-            if (operationProbes != null) {
-                ((IMultipleProbesWorker) worker).setProbeMap(operationProbes);
-            }
-            spawner.spawn(new WorkerTask(worker));
+        if (runStrategies.size() > 1) {
+            //todo: better exception
```

Comment:
Can't we make a single check `runStrategies.size() != 1` with a single message like `A Test has to contain exactly one run strategy, which is a method annotated with %s, %s or %s`?